File: angular/src/app/model/matiere.ts
import {Formateur} from './RessourceHumaineHeritage/formateur';
import {Lesson} from './lesson';

export class Matiere {
  constructor(private _id?: number, private _titre?: string, private _niveau?: string, private _duree?: number, private _objectifs?: string,
              private _prerequis?: string, private _contenu?: string, private _formateurs?: Formateur[], private _lessons?: Lesson[]) {
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get titre(): string {
    return this._titre;
  }

  set titre(value: string) {
    this._titre = value;
  }

  get niveau(): string {
    return this._niveau;
  }

  set niveau(value: string) {
    this._niveau = value;
  }

  get duree(): number {
    return this._duree;
  }

  set duree(value: number) {
    this._duree = value;
  }

  get objectifs(): string {
    return this._objectifs;
  }

  set objectifs(value: string) {
    this._objectifs = value;
  }

  get prerequis(): string {
    return this._prerequis;
  }

  set prerequis(value: string) {
    this._prerequis = value;
  }

  get contenu(): string {
    return this._contenu;
  }

  set contenu(value: string) {
    this._contenu = value;
  }

  get formateurs(): Formateur[] {
    return this._formateurs;
  }

  set formateurs(value: Formateur[]) {
    this._formateurs = value;
  }

  get lessons(): Lesson[] {
    return this._lessons;
  }

  set lessons(value: Lesson[]) {
    this._lessons = value;
  }
}

File: java/src/main/java/com/sopra/Factory/model/Matiere.java
package com.sopra.Factory.model;

import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Version;

import com.fasterxml.jackson.annotation.JsonView;
import com.sopra.Factory.model.view.JsonViews;

@Entity
@Table(name = "matiere")
@SequenceGenerator(name = "seqMatiere", sequenceName = "seq_matiere", initialValue = 1, allocationSize = 1)
public class Matiere {

	@Id
	@Column(name = "id_matiere")
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqMatiere")
	@JsonView(JsonViews.Common.class)
	private Integer id;

	@Version
	private Integer version;

	@Enumerated(EnumType.STRING)
	@Column(name = "niveau")
	@JsonView(JsonViews.Common.class)
	private Profil niveau;
	
	@Column(name = "titre")
	@JsonView(JsonViews.Common.class)
	private String titre;

	@Column(name = "duree")
	@JsonView(JsonViews.Common.class)
	private Integer duree;

	@Column(name = "objectif")
	@JsonView(JsonViews.Common.class)
	private String objectif;

	@Column(name = "prerequis")
	@JsonView(JsonViews.Common.class)
	private String prerequis;

	@Column(name = "contenu")
	@JsonView(JsonViews.Common.class)
	private String contenu;

	@OneToMany(mappedBy = "matiere", fetch = FetchType.LAZY)
	@JsonView(JsonViews.Common.class)
	private Set<Lesson> lessons;

	@OneToMany(mappedBy = "key.matiere")
	private Set<MatiereFormateur> formateurs;

	public Matiere() {
		super();
	}
	
	public String getTitre() {
		return titre;
	}

	public void setTitre(String titre) {
		this.titre = titre;
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public Integer getVersion() {
		return version;
	}

	public void setVersion(Integer version) {
		this.version = version;
	}

	public Profil getNiveau() {
		return niveau;
	}

	public void setNiveau(Profil niveau) {
		this.niveau = niveau;
	}

	public Integer getDuree() {
		return duree;
	}

	public void setDuree(Integer duree) {
		this.duree = duree;
	}

	public String getObjectif() {
		return objectif;
	}

	public void setObjectif(String objectif) {
		this.objectif = objectif;
	}

	public String getPrerequis() {
		return prerequis;
	}

	public void setPrerequis(String prerequis) {
		this.prerequis = prerequis;
	}

	public String getContenu() {
		return contenu;
	}

	public void setContenu(String contenu) {
		this.contenu = contenu;
	}

	public Set<Lesson> getLessons() {
		return lessons;
	}

	public void setLessons(Set<Lesson> lessons) {
		this.lessons = lessons;
	}

	public Set<MatiereFormateur> getFormateurs() {
		return formateurs;
	}

	public void setFormateurs(Set<MatiereFormateur> formateurs) {
		this.formateurs = formateurs;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Matiere other = (Matiere) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

}
